xwiki-platform-core/xwiki-platform-display/xwiki-platform-display-html/src/main/java/org/xwiki/rendering/display/html/internal/DocumentReferenceHTMLDisplayer.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.display.html.internal;
+
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.reference.DocumentReference;
+
+/**
+ * Implementation of {@code HTMLDisplayer} for @{@code DocumentReference}.
+ *
+ * @version $Id$
+ * @since 10.10RC1
+ */
+@Component
+@Singleton
+public class DocumentReferenceHTMLDisplayer extends AbstractHTMLDisplayer<DocumentReference>
+{
+    @Override
+    public String getTemplateName()
+    {
+        return "html_displayer_document_reference.vm";
```

Comment:
The logic of the name might be moved in AbstractHTMLDisplayer in most cases probably: something like `html_displayer/<short class name>.vm` (to discuss) for example (my second comment being that it would be much nicer to use path instead of long file names). And only override getTemplateName() for types not happy with the default logic.